applications/GeoMechanicsApplication/custom_constitutive/coulomb_with_tension_cut_off_impl.cpp

```diff
@@ -104,51 +46,106 @@ bool CoulombWithTensionCutOffImpl::IsAdmissibleSigmaTau(const Vector& rTrialSigm
+    for (auto counter = std::size_t{0}; counter < mMaxNumberOfPlasticIterations; ++counter) {
+        if (IsStressAtTensionApexReturnZone(rTrialSigmaTau)) {
+            return ReturnStressAtTensionApexReturnZone();
         }
 
-        const auto corner_point = CalculateCornerPoint(mCoulombYieldSurface.GetFrictionAngleInRadians(),
-                                                       mCoulombYieldSurface.GetCohesion(),
-                                                       mTensionCutOff.GetTensileStrength(), apex);
-
-        if (IsStressAtTensionCutoffReturnZone(rTrialSigmaTau, mTensionCutOff.GetTensileStrength(),
-                                              apex, corner_point)) {
-            return ReturnStressAtTensionCutoffReturnZone(
-                rTrialSigmaTau, mTensionCutOff.DerivativeOfFlowFunction(rTrialSigmaTau),
-                mTensionCutOff.GetTensileStrength());
+        if (IsStressAtTensionCutoffReturnZone(rTrialSigmaTau)) {
+            return ReturnStressAtTensionCutoffReturnZone(rTrialSigmaTau);
         }
 
-        if (IsStressAtCornerReturnZone(
-                rTrialSigmaTau, mCoulombYieldSurface.DerivativeOfFlowFunction(rTrialSigmaTau, AveragingType),
-                corner_point)) {
-            result = corner_point;
+        if (IsStressAtCornerReturnZone(rTrialSigmaTau, AveragingType)) {
+            result = CalculateCornerPoint();
         } else { // Regular failure region
-            result = ReturnStressAtRegularFailureZone(
-                rTrialSigmaTau, mCoulombYieldSurface,
-                mCoulombYieldSurface.DerivativeOfFlowFunction(rTrialSigmaTau, AveragingType));
+            result = ReturnStressAtRegularFailureZone(rTrialSigmaTau, AveragingType);
         }
 
-        const auto lambda = mCoulombYieldSurface.CalculatePlasticMultiplier(
-            rTrialSigmaTau, mCoulombYieldSurface.DerivativeOfFlowFunction(rTrialSigmaTau, AveragingType));
-        const double kappa = kappa_start + mCoulombYieldSurface.CalculateEquivalentPlasticStrain(
-                                               rTrialSigmaTau, AveragingType, lambda);
+        const auto kappa = kappa_start + mCoulombYieldSurface.CalculateEquivalentPlasticStrainIncrement(
+                                             rTrialSigmaTau, AveragingType);
         mCoulombYieldSurface.SetKappa(kappa);
 
-        double error = std::abs(mCoulombYieldSurface.YieldFunctionValue(result));
-        if (error < mCoulombYieldSurface.GetConvergenceTolerance()) break;
+        if (std::abs(mCoulombYieldSurface.YieldFunctionValue(result)) < mAbsoluteYieldFunctionValueTolerance) {
+            break;
+        }
     }
     return result;
 }
 
+Vector CoulombWithTensionCutOffImpl::CalculateCornerPoint() const
+{
+    // Check whether the tension cut-off lies beyond the apex
+    auto result                 = Vector{ZeroVector(2)};
+    result[0]                   = mCoulombYieldSurface.CalculateApex();
```

Comment:
We can write this more compact (and perhaps slightly clearer) in the following way:
```suggestion
    auto result                 = UblasUtilities::CreateVector({mCoulombYieldSurface.CalculateApex(), 0.0});
```